File: PathfinderCombatSimulator.ClassLibrary/TransferObjects/CombatTeam.cs
namespace PathfinderCombatSimulator
{
    public class CombatTeam
    {
        public CombatTeam(string acgName)
        {
            Name = acgName;
        }

        public string Name { get; set; }

        public List<Mob> Combatants { get; set; } = new List<Mob>();

        public bool Contains(Mob contained)
        {
            foreach(Mob mob in Combatants)
            {
                if (mob == contained) return true;
            }
            return false;
        }
            
    }
}

File: CPlusPlus/PathfinderCombatSimulator.ConsoleCpp/combat_team.cpp
#include "types.h"

using namespace pathfinder_combat_simulator;

[[nodiscard]] bool combat_team::contains(nullable<mobile_object> contained) const
{
	for (nullable<mobile_object> mob : combatants)
	{
		if (mob == contained) return true;
	}
	return false;
}
